src/ode_solve.jl

```diff
@@ -252,6 +271,37 @@ function inner_loss(phi::ODEPhi{C, T, U}, f, autodiff::Bool, t::AbstractVector,
     sum(abs2, dxdtguess .- fs) / length(t)
 end
 
+"""
+L2 inner loss for DAEProblems
+"""
+
+function inner_loss_DAE end
+
+function inner_loss_DAE(phi::ODEPhi{C, T, U}, f, autodiff::Bool, t::Number, θ,
+                    p, u) where {C, T, U <: Number}
+    sum(abs2, ode_dfdx(phi, t, θ, autodiff) - f(phi(t, θ), u, p, t))
```

Comment:
u is phi. This is completely off. Please try writing this on pen and paper first.

Reply:
will do, got some stuff mixed up